algorithms/a3c/model.py

```diff
@@ -12,22 +13,45 @@
 import torch.nn.functional as F
 
 
-def calculate_lstm_input_size_after_4_conv_layers(frame_dim, stride=2, kernel_size=3, padding=1,
+def calculate_lstm_input_size_for_A3C(resolution, stride=2, kernel_size=3, padding=1,
                                      num_filters=32):
     """
-    Assumes square resolution image. Find LSTM size after 4 conv layers below in A3C using regular
-    Convolution math. For example:
+    Find LSTM size after 4 conv layers below in A3C using regular
+    convolution math. For example:
     42x42 -> (42 − 3 + 2)÷ 2 + 1 = 21x21 after 1 layer
     11x11 after 2 layers -> 6x6 after 3 -> and finally 3x3 after 4 layers
     Therefore lstm input size after flattening would be (3 * 3 * num_filters)
     """
-
-    width = (frame_dim - kernel_size + 2 * padding) // stride + 1
+    width = (resolution[0] - kernel_size + 2 * padding) // stride + 1
     width = (width - kernel_size + 2 * padding) // stride + 1
     width = (width - kernel_size + 2 * padding) // stride + 1
     width = (width - kernel_size + 2 * padding) // stride + 1
 
-    return width * width * num_filters
+    height = (resolution[1] - kernel_size + 2 * padding) // stride + 1
+    height = (height - kernel_size + 2 * padding) // stride + 1
+    height = (height - kernel_size + 2 * padding) // stride + 1
+    height = (height - kernel_size + 2 * padding) // stride + 1
+
+    return width * height * num_filters
+
+def calculate_input_width_height_for_A3C_LSTM_GA(resolution):
+    """
+    Does't assume square resolution image (since ViZDoom default is not square).
+    Similar to the calculate_lstm_input_size_for_A3C function except that there are only
+    3 conv layers and there is variation among the kernel_size, stride, the number of channels
+    and there is no padding. Therefore these are hardcoded. Check A3C_LSTM_GA class for these numbers.
+    Also, returns tuple representing (width, height, num_output_filters) instead of size
+    """
+
+    width = (resolution[0] - 8) // 4 + 1
+    width = (width - 4) // 2 + 1
+    width = (width - 4) // 2 + 1
+
+    height = (resolution[1] - 8) // 4 + 1
+    height = (height - 4) // 2 + 1
+    height = (height - 4) // 2 + 1
+
+    return width, height, 64
 
 def normalized_columns_initializer(weights, std=1.0):
```

Comment:
Why do I need column normalization. Add an explanation here

Reply:
Also ellaborate more on this particular type of normalization being used as opposed to other ones